### PR TITLE
Adds Vapes

### PR DESCRIPTION
**Pull request by RFultra**
This PR adds a vape that can be got from the cigarette machine contraband
it can be loaded with a beaker and will put the chem into your body when you vape
also can be emagged to explode in people face

### CODE REVIEW

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 1]
```diff
@@ -761,3 +761,79 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 		lit = 0
 		update_brightness()
 		visible_message("<span class='warning'>Without warning, the flame on \the [src] suddenly goes out in a weak fashion.</span>")
+
+/obj/item/weapon/vape
+	name = "E-Cigarette"
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/cigs_lighters.dm]: `\improper`

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 2]
```diff
@@ -761,3 +761,79 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+	desc = "More commony refered to as a mouth fedora"
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Misspelled commonly and referred. Also missing a period.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 3]
```diff
@@ -761,3 +761,79 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+	icon = 'icons/obj/items.dmi'
+	icon_state = "vape"
+	var/datum/effect/effect/system/smoke_spread/smoke = new()
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/cigs_lighters.dm]: You can't call procs in an item definition.

**Comment by Commandersand** [comment]
oh shit nigger what are you doing

**Comment by Exxion** [comment]
You sure are memeing pretty fucking hard for someone who just got off a repoban

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 2 above)
Comment (Exxion) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Missing an l and a period

Comment (Exxion) [reply code/game/objects/items/weapons/cigs_lighters.dm]: And an r. Wowee

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 4]
```diff
@@ -761,3 +761,79 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+			to_chat(user, "<span class='warning'>That beaker is too large to fit into \the [src]'s beaker port.</span>")
+			return
+		if(!user.drop_item(W, src))
+			to_chat(user, "<span class='warning'>You can't let go of \the [W]!</span>")
+			return 1
+		beaker = W
+		to_chat(user, "You insert \the [W] into \the [src]'s beaker port.")
+
+/obj/item/weapon/vape/AltClick(mob/user)
+	if(usr.incapacitated())
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/cigs_lighters.dm]: This should be `user`, since you have access to it.

**Comment by ghost** [comment]
we should have a server vote to have vaping valid tbh

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 5]
```diff
@@ -761,3 +761,79 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+		var/obj/item/weapon/reagent_containers/glass/beaker/B = beaker
+		B.forceMove(user.loc)
+		user.put_in_hands(B)
+		beaker = null
+		to_chat(user, "You remove \the [B] from \the [src].")
+
+/obj/item/weapon/vape/attackby(obj/item/weapon/W, mob/user)
+	if(istype(W, /obj/item/weapon/card/emag))
+		emagged = TRUE
+	else return
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/cigs_lighters.dm]: This is redundant, you can simply remove this line.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 6]
```diff
@@ -761,3 +761,79 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+		user.apply_damage(20, BURN, "head")
+		qdel(src)
+		var/datum/effect/effect/system/spark_spread/sparks = new /datum/effect/effect/system/spark_spread()
+		sparks.set_up(3, 0, get_turf(user))
+		sparks.start()
+		to_chat(user, "<span class='userdanger'>The [name] suddenly explodes in your mouth!</span>")
+	else
+
+		var/obj/item/weapon/reagent_containers/glass/beaker/B = beaker
+		user.delayNextAttack(50)
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/cigs_lighters.dm:806]: Have you tested this? I'm not certain that `delayNextAttack()` delays `attack_self()` calls.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 7]
```diff
@@ -761,3 +761,79 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 		lit = 0
 		update_brightness()
 		visible_message("<span class='warning'>Without warning, the flame on \the [src] suddenly goes out in a weak fashion.</span>")
+
+/obj/item/weapon/vape
+	name = "E-Cigarette"
+	desc = "More commonly referred to as a mouth fedora."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "vape"
+	var/datum/effect/effect/system/smoke_spread/smoke = new()
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/cigs_lighters.dm]: You can't call procs in an item definition.

**Review by Shadowmech88 — CHANGES_REQUESTED** [review]
Some things need correcting.

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 7 above)
Comment (Exxion) [reply code/game/objects/items/weapons/cigs_lighters.dm]: I'm pretty sure `new` is an exception

Comment (Shadowmech88) [reply code/game/objects/items/weapons/cigs_lighters.dm]: Why isn't Travis building this? The outcome of the compile would answer the question.

Comment (gbasood) [reply code/game/objects/items/weapons/cigs_lighters.dm]: this is fine

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 4 above)
Comment (gbasood) [reply code/game/objects/items/weapons/cigs_lighters.dm]: don't ever user `usr` for the record

Comment (Shadowmech88) [reply code/game/objects/items/weapons/cigs_lighters.dm]: Isn't it necessary in some places, like verbs?

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 6 above)
Comment (RFultra) [reply code/game/objects/items/weapons/cigs_lighters.dm:806]: worked for me

**Comment by Sailol** [comment]
A decent meme, polyacid smoke vapes when?

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 4 above)
Comment (Exxion) [reply code/game/objects/items/weapons/cigs_lighters.dm]: Strictly speaking, yes, but the settings don't work the same way as other code, and it's not necessary outside them

**Comment by Orangerush** [comment]
I know it's just a meme but it's a neat idea that you can fill it with chems of your choice. Like a better version of cig dipping

**Comment by RFultra** [comment]
\improper ?

**Comment by Shadowmech88** [comment]
Instead of `E-Cigarette` it would be `\improper E-Cigarette`.
It prevents the item from being referred to as a proper noun.

**Comment by Intigracy** [comment]
This might be a meme but as Orangerush said it's a decent idea.

**Comment by Shadowmech88** [comment]
I didn't say that.

**Comment by Fruktlimpa** [comment]
Sprite looks extremely basic, is there no way to give it some flair?

**Comment by RFultra** [comment]
how do I fix conflicts?

**Comment by ghost** [comment]
depends, what are you using for your git?

Pull from pomf bleeding-edge and rebase, then push is usually the way to do it

**Comment by RFultra** [comment]
I think I fixed it but I dunno

**Comment by PJB3005** [comment]
Yeah it's fixed.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 8]
```diff
@@ -776,4 +774,81 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+			B.reagents.trans_to(user, vape_amount)
+
+		user.visible_message("<span class='rose'>[user] takes a hit from the \the [src] and makes a nice cloud!</span>")
+		playsound(get_turf(src), 'sound/effects/smoke.ogg', 50, 1, -3)
+		smoke.set_up(1, 0, src.loc)
+		smoke.attach(src)
+		smoke.start()
+
+/obj/item/weapon/vape/suicide_act(mob/user)
+	to_chat(viewers(user), "<span class='danger'>[user] is going crazy on the [src.name]! It looks like \he's  trying to commit suicide!</span>")
```
Comment (PJB3005) [inline code/game/objects/items/weapons/cigs_lighters.dm:818]: Use `visible_message` here.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 9]
```diff
@@ -776,4 +774,81 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+////////
+
+/obj/item/weapon/vape
+	name = "\improper E-Cigarette"
+	desc = "More commonly referred to as a mouth fedora"
+	icon = 'icons/obj/items.dmi'
+	icon_state = "vape"
+	var/datum/effect/effect/system/smoke_spread/smoke = new()
+	var/max_beaker_volume = 50
+	var/beaker = null
```
Comment (PJB3005) [inline code/game/objects/items/weapons/cigs_lighters.dm:790]: Cast this to a beaker always so you won't have to cast it in the attack self code.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 10]
```diff
@@ -776,4 +774,81 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+		if(B.volume > max_beaker_volume)
+			to_chat(user, "<span class='warning'>That beaker is too large to fit into \the [src]'s beaker port.</span>")
+			return
+		if(!user.drop_item(W, src))
+			to_chat(user, "<span class='warning'>You can't let go of \the [W]!</span>")
+			return 1
+		beaker = W
+		to_chat(user, "You insert \the [W] into \the [src]'s beaker port.")
+	if(istype(W, /obj/item/weapon/card/emag))
+		emagged = TRUE
```
Comment (PJB3005) [inline code/game/objects/items/weapons/cigs_lighters.dm:840]: Give it some feedback.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 11]
```diff
@@ -776,4 +774,81 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+
+
+/obj/item/weapon/vape/AltClick(mob/user)
+	if(user.incapacitated())
+		return
+	if(!beaker)
+		return
+	else
```
Comment (PJB3005) [inline code/game/objects/items/weapons/cigs_lighters.dm:848]: Don't need an else here.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 12]
```diff
@@ -776,4 +774,81 @@ MATCHBOXES ARE ALSO IN FANCY.DM
...
+		if(!user.drop_item(W, src))
+			to_chat(user, "<span class='warning'>You can't let go of \the [W]!</span>")
+			return 1
+		beaker = W
+		to_chat(user, "You insert \the [W] into \the [src]'s beaker port.")
+	if(istype(W, /obj/item/weapon/card/emag))
+		emagged = TRUE
+
+
+/obj/item/weapon/vape/AltClick(mob/user)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/cigs_lighters.dm:843]: Having only alt click as ejection method seems cumbersome.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by yclatious** [comment]
>upvoting your own PR
>adding hand fedoras

Waifufags, not even once.

**Comment by Fruktlimpa** [comment]
Honestly, upvoting your own PR should be an instant repoban. Only a barbarian does that.

**Comment by ghost** [comment]
the memes jack

**Comment by SonixApache** [comment]
fuck off

**Comment by Shadowmech88** [comment]
The sprite could use some shading.

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 12 above)
Comment (RFultra) [reply code/game/objects/items/weapons/cigs_lighters.dm:843]: So should I have a right click verb also?

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 9 above)
Comment (RFultra) [reply code/game/objects/items/weapons/cigs_lighters.dm:790]: I am not sure what you mean here

**Comment by ghost** [comment]
Hippie-tier meme

That aside, mechanically speaking: The reagent transfer is exactly the same as holding the beaker in your hand and drinking from it. Nothing interesting there, so this thing is just an infinite-use handheld smoke machine (that you can get for free by pulsing a wire). Both lame and kind of OP.

**Comment by RFultra** [comment]
the amount of smoke it makes is minuscule

**Comment by ghost** [comment]
If only there were some pictures in the OP

**Pull request closed without merging**